### DISCUSSION TITLE
Problema para Salvar - Editor HTML

### DISCUSSION THREAD

**Discussion by Fudashow** [balta-io/forum#293] [Tira Dúvidas] [answered]
Pessoal, boa tarde ! 

Estou tendo o seguinte erro no meu código "O nome text não existe no contexto atual" poderiam me ajudar

Editor.cs: 

using System;
using System.Text;
using System.IO;

namespace EditorHtml {

        public static class Editor {

            public static void Show(){
                Console.Clear();
                Console.BackgroundColor = ConsoleColor.White;
                Console.ForegroundColor = ConsoleColor.Black;
                Console.Clear();
                Console.WriteLine("MODO EDITOR");
                Console.WriteLine("----------------");
                Start();
                
            }

            public static void Start(){
                var file = new StringBuilder();
            
                do
                {
                    file.Append(Console.ReadLine());
                    file.Append(Environment.NewLine);
                } while (Console.ReadKey().Key != ConsoleKey.Escape);
                

                Console.WriteLine("----------------");
                Console.WriteLine(" 1 - Se deseja salvar o arquivo"); 
                Console.WriteLine(" 2 - Para sair");

              var escolha = short.Parse(Console.ReadLine());
                
                switch (escolha)
                {
                    case 1: Salvar(text); break;
                    case 2: Environment.Exit(0);break;
                    default: ;break;
                    
                }

            }

           public static void Salvar(string text){
                
                Console.Clear();
                Console.WriteLine("Qual caminho deseja salvar o arquivo? ");
                var path = Console.ReadLine();

                using (var file = new StreamWriter(path))
                {
                    file.Write(text);
                }

                Console.WriteLine($" Arquivo {path} salvo com sucesso! ");
                Menu.Show();
            }
            }
        }

**Comment by PriciLeite** [comment]
Parece ser um erro de inicialização da variavel... 

var escolha = short.Parse(Console.ReadLine());
 string text = null;           
            switch (escolha)
            {
                case 1: Salvar(text); break;
                case 2: Environment.Exit(0);break;
                default: ;break;
                
            }

testaria inicializando com null...

**Comment by v1nifelix** [comment, marked as answer]
Parece que você não está inicializando e nem declarando a varivel ``` text ```

utilize:

```

public static void Start(){
    var text = new StringBuilder();
    do
    {
        text.Append(Console.ReadLine());
        text.Append(Environment.NewLine);
    } while (Console.ReadKey().Key != ConsoleKey.Escape);
            
    Console.WriteLine("----------------");
    Console.WriteLine(" 1 - Se deseja salvar o arquivo"); 
    Console.WriteLine(" 2 - Para sair");

    var escolha = short.Parse(Console.ReadLine());
            
    switch (escolha)
    {
        case 1: Salvar(text.ToString()); break;
        case 2: Environment.Exit(0);break;
        default: ;break;
    }
}


```

**Reply by Fudashow** [reply]
> Parece que você não está inicializando e nem declarando a variável`text`
> 
> utilizar:
> 
> ```
> 
> public static void Start(){
>     var text = new StringBuilder();
>     do
>     {
>         text.Append(Console.ReadLine());
>         text.Append(Environment.NewLine);
>     } while (Console.ReadKey().Key != ConsoleKey.Escape);
>             
>     Console.WriteLine("----------------");
>     Console.WriteLine(" 1 - Se deseja salvar o arquivo"); 
>     Console.WriteLine(" 2 - Para sair");
> 
>     var escolha = short.Parse(Console.ReadLine());
>             
>     switch (escolha)
>     {
>         case 1: Salvar(text.ToString()); break;
>         case 2: Environment.Exit(0);break;
>         default: ;break;
>     }
> }
> ```

Valeu @V1NIFELIX me ajudou bastante.
